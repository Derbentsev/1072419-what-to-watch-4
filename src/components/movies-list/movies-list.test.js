import MoviesList from './movies-list';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';


const films = [
  {
    title: `title1`,
    poster: `image1`,
    src: `src1`,
  },
  {
    title: `title2`,
    poster: `image2`,
    src: `src2`,
  },
  {
    title: `title3`,
    poster: `image3`,
    src: `src3`,
  },
];

const mockStore = configureStore([]);

it(`Render Movies List`, () => {
  const store = mockStore({
    activeFilm: null,
    films,
    showedFilmsCount: 8,
    currentFilter: `filter1`,
  });

  const tree = renderer
    .create(
        <Provider store = {store}>
          <MoviesList
            films = {films}
            handleMovieCardMouseEnter = {() => {}}
          />
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
