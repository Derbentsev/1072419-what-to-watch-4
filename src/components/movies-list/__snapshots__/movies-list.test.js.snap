// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movies List 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="title1"
        height="175"
        src="img/image1"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        title1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="title1"
        height="175"
        src="img/image1"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        title1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="title1"
        height="175"
        src="img/image1"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        title1
      </a>
    </h3>
  </article>
</div>
`;
